kernel/fs/iso9660.inc

```diff
@@ -736,3 +736,1044 @@ cd_compare_name:
+        cmp     ax, 0x3B00
+        je      @f
+        xchg    ah, al
+        call    UTF16to8
+        dec     ebx
+        jnz     @b
+@@:
+        mov     byte[edi], 0
+;-----------------------------------------------------------------------------
+.end_copy:
+        pop     eax
+        pop     edi
+        pop     ebx
+        add     edi, 264
+        cmp     eax, 2
+        jb      .skip
+        add     edi, 520-264
+.skip:
+        movzx   ecx, byte[ebx]
+        add     ebx, ecx
+
+        test    ebx, CDBlockSize
+        jnz     .next_sector
+
+        mov     eax, ebx
+        and     eax, not -CDBlockSize
+        cmp     eax, [esp + 4]
+        jb      .new_file
+
+        mov     eax, [esp + 12]
+        cmp     eax, [esp + 8]
+        jb      .new_file
+
+        jmp     .end_loop
+
+.next_sector:
+        inc     dword[esp]
+        sub     dword[esp + 4], CDBlockSize
+        ja      .read_sector
+.end_loop:
+        mov     ecx, [esp + 28]
+        mov     ebx, [esp + 12]
+        mov     [ecx + 4], ebx
+        mov     esi, [esp + 16]
+        mov     [ecx + 8], esi
+        mov     esi, [esp + 8] ; max count
+        ; free buffer
+        add     esp, 8*4
+        call    kernel_free
+
+        xor     eax, eax
+        cmp     ebx, esi
+        je      @f
+        mov     eax, EEOF
+@@:
+        ret
+.err_disk:
+        mov     ecx, [esp + 28]
+        mov     ebx, [esp + 12]
+        mov     [ecx + 4], ebx
+        mov     esi, [esp + 16]
+        mov     [ecx + 8], esi
+        ; free buffer
+        add     esp, 8*4
+        call    kernel_free
+
+        mov     eax, EIO
+        ret
+
+;-----------------------------------------------------------------------------
+; ISO9660 external functions
+;   in:
+; ebx -> parameter structure of sysfunc 70
+; ebp -> ISO9660 structure
+; esi -> path string in UTF-8
+;   out:
+; eax, ebx = return values for sysfunc 70
+iso9660_GetFileInfo:
+        cmp     byte[esi], 0
+        jz      .rootdir
+
+        sub     esp, 4 ; for ptr on memory page
+        call    iso9660_find_file
+
+        mov     ecx, [ebx + 16] ; buffer
+
+        call    .copy_file_info
+
+        call    kernel_free
+        xor     eax, eax
+        mov     ebx, 40
+        ret
+
+;  IN: eax -> ISO966_DIRECTORY_RECORD
+;      ecx -> buffer
+; destruct: edx
+.copy_file_info:
+         ; copy size
+        mov     [ecx + 36], dword 0
+        mov     edx, [eax + ISO9660_DIRECTORY_RECORD.data_length]
+        mov     [ecx + 32], edx
+
+        ; copy flags(dir of file)
+        xor     edx, edx
+        or      dl, 000001b
+        test    byte[eax + ISO9660_DIRECTORY_RECORD.flags], 1b ; check hidden flag
+        jz      @f
+        or      dl, 10b
+@@:
+        test    byte[eax + ISO9660_DIRECTORY_RECORD.flags], 10b ; check dir
+        jz      @f
+        or      dl, 10000b ; dir flag
+        mov     dword[ecx + 32], 0 ; size = zero
+@@:
+        mov     [ecx], edx
+
+        ; copy date creat file
```

Comment:
you mean file creation date? correct please

Reply:
fixed